src/persist-client/src/operators/shard_source.rs

```diff
@@ -66,27 +67,28 @@ use crate::{PersistLocation, ShardId};
 /// using [`timely::dataflow::operators::generic::operator::empty`].
 ///
 /// [advanced by]: differential_dataflow::lattice::Lattice::advance_by
-pub fn shard_source<K, V, D, F, G>(
+pub fn shard_source<T, K, V, D, F, G>(
     scope: &mut G,
     name: &str,
     clients: Arc<PersistClientCache>,
     location: PersistLocation,
     shard_id: ShardId,
-    as_of: Option<Antichain<G::Timestamp>>,
-    until: Antichain<G::Timestamp>,
+    as_of: Option<Antichain<T>>,
+    until: Antichain<T>,
     flow_control: Option<FlowControl<G>>,
```

Comment:
i'd love to make the naming be a bit more precise on how these two relate, if we think they're both sticking around. maybe something like ts_flow_control and part_flow_control?